src/components/Meeting.tsx

```diff
@@ -26,16 +26,28 @@ import Link from './Link';
 import Map from './Map';
 
 import { useData, useFilter, useInput, useSettings } from '../hooks';
+import { useParams } from 'react-router-dom';
 import type { Meeting as MeetingType } from '../types';
 import Loading from './Loading';
 
 export default function Meeting() {
+  const { slug } = useParams();
   const { capabilities, meetings } = useData();
 
   const { settings, strings } = useSettings();
   const { input } = useInput();
   const { meeting, waitingForFilter } = useFilter();
 
+  // check if we have a slug but no meeting from FilterProvider,
+  if (slug && !meeting && !waitingForFilter && Object.keys(meetings).length > 0) {
+    // Check directly if meeting exists
+    if (!(slug in meetings)) {
+      throw new Error(strings.not_found);
+    }
+    // Meeting exists in data but FilterProvider hasn't found it yet - keep waiting
+    return <Loading />;
+  }
+
```

Comment:
we've got some logic lower down that tries to do approximately this - i think we should move this down to replace that logic - it's down there so prevent any hooks from running conditionally